lib/pbench/server/indexing_tarballs.py

```diff
@@ -430,13 +466,9 @@ def sighup_handler(*args):
                             )
                             tb_res = error_code["OP_ERROR" if failures > 0 else "OK"]
                         finally:
-                            if dataset:
+                            if dataset and tb_res.success:
```

Comment:
Interesting question. The logic is complicated and I guess I was too lazy to try to analyze it when I added the success check.

However, since we set `dataset` in the *outer* `try` block, and this `finally` is attached to an **inner** `try` block, it seems reasonable to assume that `dataset` must be valid if we get here.